celery/app/task.py

```diff
@@ -919,6 +919,9 @@ def replace(self, sig):
 
         if self.request.is_eager:
             return sig.apply().get()
+            # task_result = sig.apply()
+            # with allow_join_result():
+            #     return task_result.get()
```

Comment:
This is probably what you meant.

```suggestion
            task_result = sig.apply()
            with allow_join_result():
                  return task_result.get()
```